From 849e8927764c6cce75f5c4aac36d3e0809c7bb9f Mon Sep 17 00:00:00 2001
From: Fuyan Yuan <33221728+TheRainstorm@users.noreply.github.com>
Date: Tue, 29 Oct 2024 17:39:38 +0800
Subject: [PATCH 2/2] fix cudnn8 cudnnSetRNNDescriptor error

---
 code/nvidia/cudnn_helper.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/code/nvidia/cudnn_helper.h b/code/nvidia/cudnn_helper.h
index 5ef7afe..3c1759e 100644
--- a/code/nvidia/cudnn_helper.h
+++ b/code/nvidia/cudnn_helper.h
@@ -375,7 +375,11 @@ public:
 
 
 #if CUDNN_MAJOR >= 7
+    #if CUDNN_MAJOR >= 8
+        CHECK_CUDNN_ERROR(cudnnSetRNNDescriptor_v6(cudnn_handle,
+    #else
         CHECK_CUDNN_ERROR(cudnnSetRNNDescriptor(cudnn_handle,
+    #endif
                                                 *desc,
 #else
         CHECK_CUDNN_ERROR(cudnnSetRNNDescriptor(*desc,
-- 
2.34.1

